packages/core/chrome/core-chrome-browser-internal/src/ui/header/collapsible_nav.tsx

```diff
@@ -162,35 +163,35 @@ export function CollapsibleNav({
       size={248}
     >
       {customNavLink && (
-        <Fragment>
+        <>
           <EuiFlexItem grow={false} style={{ flexShrink: 0 }}>
             <EuiCollapsibleNavGroup
               background="dark"
               className="eui-yScroll"
               style={{ maxHeight: '40vh' }}
             >
-              <EuiListGroup
-                listItems={[
-                  createEuiListItem({
-                    link: customNavLink,
-                    basePath,
-                    navigateToUrl,
-                    dataTestSubj: 'collapsibleNavCustomNavLink',
-                    onClick: closeNav,
-                    externalLink: true,
-                    iconProps: { color: 'ghost' },
-                  }),
-                ]}
-                maxWidth="none"
-                gutterSize="none"
-                size="s"
-                color="ghost"
-              />
+              <EuiThemeProvider colorMode="dark">
```

Comment:
We're already providing the theme provider at the root level of the react tree, so why do you need to add another one here? Forcing colorMode to `dark` seems wrong to me?

Reply:
It's done because these types of sections in the `EuiCollapsibleNav` is **always** dark. You can see an example on our docs here: https://elastic.github.io/eui/#/navigation/collapsible-nav#nav-groups-with-lists-and-other-content

<img width="458" alt="" src="https://user-images.githubusercontent.com/549407/201991154-12b300a7-6a13-4026-8867-3a7ca9ec3dd8.png">

The reason why switching/forcing the colorMode is superior is twofold: it cascades to **all** child components - links, text color, buttons, icon fills, etc. - everything just switches without needing a bunch of custom cascading/nesting CSS.

`EuiBottomBar` also has the same logic / nested dark provider as this.

Reply:
We also dropped support for `EuiListGroup[color=ghost]` as a design decision,

> Breaking changes
> * Removed ghost color from EuiListGroupItem. Consumers now must pass a EuiThemProvider color="dark" to use a ghost color.
> * Removed inherit color from EuiListGroupItem. The inherit in my opinion was not making sense anymore. The consumer should pass the color they think makes sense for their use case.
> * The list items now defaults to color text

and our upgrade strategy for Kibana historically defaults to _maintain the same look & feel_ unless we've pre-coordinated changes with affected teams, which is why we're adding the theme provider here instead of changing the UX away from the ghost color.

FYI this is prompting a wider discussion about these types of breaking changes in https://github.com/elastic/eui/discussions/6364

Reply:
@miukimiu mentioned this as well in our sync today, but I would be curious if people would feel less confused about this change if we preserved `color="ghost"` as a prop/API option and had the component automatically internally wrap `<EuiThemeProvider colorMode="dark">` instead of consumers explicitly having to.

Reply:
Thanks for the explanations